packages/shared/lib/core/wallet/actions/events-handlers/handleSpentOutputEvent.ts

```diff
@@ -32,14 +32,14 @@ export async function handleSpentOutputEventInternal(walletId: string, payload:
         const accountOutputs = await wallet.accounts()
         const implicitAccountOutputs = await wallet.implicitAccounts()
         updateActiveWallet(walletId, { walletOutputs, accountOutputs, implicitAccountOutputs })
-        if (
-            wallet.mainAccountId &&
-            !walletOutputs.find(
-                (output) =>
-                    output.output.type === OutputType.Account &&
-                    (output as unknown as AccountOutput).accountId === wallet.mainAccountId
-            )
-        ) {
+        const isSendingTheMainAccount = walletOutputs.find(
```

Comment:
this code here just means that you have an account that has the same ID as your main. We have to check where we can know if the account was spent & returned to you or just spent and send somwhere else or burned. We need to stufy if in this event `walletOutputs` or `accountOutputs` already has the new unspent account ot not. If not, we might no be able to handle this logic here, maybe it could be moved `handleTransactionInclusionEvent` though we should study well the events